src/sif/pedestriancost.cc

```diff
@@ -620,14 +683,14 @@ Cost PedestrianCost::EdgeCost(const baldr::DirectedEdge* edge,
   }
 
   float sec =
-      edge->length() * speedfactor_ * kSacScaleSpeedFactor[static_cast<uint8_t>(edge->sac_scale())];
+      edge->length() * speedfactor_ * kSacScaleSpeedFactor[static_cast<uint8_t>(edge->sac_scale())] * kGradeBasedSpeedFactor[static_cast<uint8_t>(edge->weighted_grade())];
```

Comment:
this is much more intuitive to me now. i still wonder t hough about the interplay between sac_scale and grade. we will pretty much have grade coverage for the whole dataset but sac scale will be sparsely populated. what i think that will mean is that any place that actually tagged sac scale will get penalized while anything untagged will get preference. does anyone else have thoughts/worries about that? basically im worried that having both is unfair to sac scale tagged edges because they are rare and we'll end up accidentally avoiding them just because hardly anything is tagged with that (though i guess this is already the case)

Reply:
Indeed, using sac_scale to penalize would be wrong. Unless plain "path" without much specifics would penalize even more.